core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMGraph.kt

```diff
@@ -158,7 +159,7 @@ class STDCMGraph(
             maxTime = min(maxTime, maxDelayAddedOnEdge)
 
             remainingDistance -= edge.length.distance
-            if (edge.beginSpeed == 0.0 || remainingDistance <= 0.meters) return maxTime
+            if (edge.beginSpeed == 0.0 || remainingDistance <= 0.meters) return max(0.0, maxTime)
```

Comment:
I'd rather you did this on `maxDelayAddedOnEdge`, it makes more sense for it to always be positive.